src/sentry/integrations/vsts/search.py

```diff
@@ -1,60 +1,37 @@
-from typing import Any
+from typing import TypeVar
 
-from rest_framework.request import Request
 from rest_framework.response import Response
 
-from sentry.api.api_owners import ApiOwner
-from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import control_silo_endpoint
-from sentry.hybridcloud.rpc import coerce_id_from
-from sentry.integrations.api.bases.integration import IntegrationEndpoint
-from sentry.integrations.models.integration import Integration
+from sentry.integrations.source_code_management.issues import SourceCodeIssueIntegration
+from sentry.integrations.source_code_management.search import SourceCodeSearchEndpoint
 from sentry.integrations.vsts.integration import VstsIntegration
-from sentry.organizations.services.organization import RpcOrganization
+
+T = TypeVar("T", bound=SourceCodeIssueIntegration)
 
 
 @control_silo_endpoint
-class VstsSearchEndpoint(IntegrationEndpoint):
-    owner = ApiOwner.UNOWNED
-    publish_status = {
-        "GET": ApiPublishStatus.PRIVATE,
-    }
-
-    def get(
-        self, request: Request, organization: RpcOrganization, integration_id: int, **kwds: Any
-    ) -> Response:
-        try:
-            integration = Integration.objects.get(
-                organizationintegration__organization_id=coerce_id_from(organization),
-                id=integration_id,
-                provider="vsts",
-            )
-        except Integration.DoesNotExist:
-            return Response(status=404)
-
-        field = request.GET.get("field")
-        query = request.GET.get("query")
-        if field is None:
-            return Response({"detail": "field is a required parameter"}, status=400)
+class VstsSearchEndpoint(SourceCodeSearchEndpoint):
+    @property
+    def integration_provider(self):
+        return "vsts"
+
+    @property
+    def installation_class(self):
+        return VstsIntegration
+
+    def handle_search_issues(self, installation: T, query: str, repo: str | None) -> Response:
         if not query:
-            return Response({"detail": "query is a required parameter"}, status=400)
-
-        installation = integration.get_installation(organization.id)
-        assert isinstance(installation, VstsIntegration), installation
-
-        if field == "externalIssue":
-            if not query:
-                return Response([])
-
-            resp = installation.search_issues(query=query)
```

Comment:
vsts doesn't query issues with repository, needed to account for that. there isn't a test for VSTS, may need to add that?

Reply:
Yeah makes sense. Please at least add barebone tests.

Reply:
done!